### PR TITLE
Preliminary work on -joystick_map redesign, try 2 for Vas review

### PR DESCRIPTION
**Pull request by Firehawke**

### CODE REVIEW

**Comment by rb6502** [comment]
I think this is a good explanation and reads well.  @cuavas can you comment since he requested you look at it?

**Comment by cuavas** [comment]
I need to find some time to finalise the artwork, sorry.

File: docs/source/commandline/commandline-all.rst [hunk 1]
```diff
@@ -1931,33 +1931,63 @@ Core Input Options
 
 **-joystick_map** *<map>* / **-joymap** *<map>*
 
-    Controls how joystick values map to digital joystick controls. MAME accepts
-    all joystick input from the system as analog data.  For true analog
-    joysticks, this needs to be mapped down to the usual 4-way or 8-way digital
-    joystick values.  To do this, MAME divides the analog range into a 9x9 grid.
-    It then takes the joystick axis position (for X and Y axes only), maps it to
-    this grid, and then looks up a translation from a joystick map.  This
+    Controls how joystick values map to digital joystick controls.
```
Comment (cuavas) [inline docs/source/commandline/commandline-all.rst]: This should explicitly say that it applies to *analog* joysticks – it doesn’t affect digital joytsticks.

File: docs/source/commandline/commandline-all.rst [hunk 2]
```diff
@@ -1931,33 +1931,63 @@ Core Input Options
...
+    
+    Systems such as Pac-Man use a 4-way joystick and will actually exhibit
+    undesired behavior when a diagonal is triggered; in the case of Pac-Man,
+    movement will stop completely at intersections when diagonals are triggered
+    and the game will be considerably harder to play correctly.  Many other
+    arcade cabinets used 4-way or 8-way joysticks (as opposed to full analog
+    joysticks), so for PC-style true analog joysticks this then needs to be
+    mapped down to the expected 4-way or 8-way digital joystick values.  This
+    is because MAME accepts all joystick input from your controller as analog
+    data.
```
Comment (cuavas) [inline docs/source/commandline/commandline-all.rst]: I don't particularly like the wording here.  "PC-style true analog joysticks" may have been a thing back when the 15-pin game port was in use, but these days there isn't really such a thing as a "PC-style joystick" at all.  Say something like "…so for true analog joysticks, including flight sticks and analog thumb sticks,…" or something.  Also the last sentence is pointless.  MAME doesn't accept *all* joystick input as analog data – it accepts analog axis values as analog data, and saying so is redundant.

File: docs/source/commandline/commandline-all.rst [hunk 3]
```diff
@@ -1931,33 +1931,63 @@ Core Input Options
...
+
     The default is ``auto``, which means that a standard 8-way, 4-way, or 4-way
     diagonal map is selected automatically based on the input port configuration
     of the current system.
 
-    Maps are defined as a string of numbers and characters. Since the grid is
-    9x9, there are a total of 81 characters necessary to define a complete map.
-    Below is an example map for an 8-way joystick:
+    Generally you will want to set up the **-joystick_map** setting in the
+    per-system ``<system>.cfg`` file as opposed to the main ``MAME.INI``
```
Comment (cuavas) [inline docs/source/commandline/commandline-all.rst]: It doesn't go in the `.cfg` file – it goes in the system-specific `.ini` file.

File: docs/source/commandline/commandline-all.rst (hunk 1 above)
Comment (Firehawke) [reply docs/source/commandline/commandline-all.rst]: Yeah, missed that on the initial pass. Some of this comes from the original text that was in there. Fixed.

File: docs/source/commandline/commandline-all.rst (hunk 2 above)
Comment (Firehawke) [reply docs/source/commandline/commandline-all.rst]: Corrected.

File: docs/source/commandline/commandline-all.rst (hunk 3 above)
Comment (Firehawke) [reply docs/source/commandline/commandline-all.rst]: Fixed.

**Comment by Firehawke** [comment]
@cuavas When you have some free time, please take a look over the changes and let me know what I need to do from here.

**Pull request closed without merging**